### PR TITLE
osemgrep: fix Metrics_.is_enabled

### PR DESCRIPTION
**Pull request by aryx**
We were just looking whether it was Off, but if it's Auto
we also need to make sure the user was using the registry (or the App)

test plan:
make e2e

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Comment by aryx** [comment]
Also fixed a few bugs in the previous implementation

File: src/osemgrep/core/Metrics_.mli [hunk 1]
```diff
@@ -1,9 +1,10 @@
 (*
    Configures metrics upload.
 
-   On - Metrics always sent
-   Off - Metrics never sent
-   Auto - Metrics only sent if config is pulled from the server
+   On   - Metrics always sent
+   Off  - Metrics never sent
+   Auto - Metrics only sent if config is pulled from the registry
```
Comment (zzeleznick) [inline src/osemgrep/core/Metrics_.mli:6]: thought – should usage of `--pro` constitute using the app?

File: src/osemgrep/core/Metrics_.ml [hunk 2]
```diff
@@ -80,18 +82,23 @@ open Semgrep_metrics_t
...
 
-     ON - Metrics always sent
-     OFF - Metrics never sent
-     AUTO - Metrics only sent if config is pulled from the server
+     On   - Metrics always sent
+     Off  - Metrics never sent
+     Auto - Metrics only sent if config is pulled from the registry
+            or if using the Semgrep App.
+
+  What is the rational for Auto? I guess if the user is requesting rules
```
Comment (zzeleznick) [inline src/osemgrep/core/Metrics_.ml:99]: I'm also for removing auto and making `metrics-off` / `disable-metrics` a flag

File: src/osemgrep/cli/CLI.ml [hunk 3]
```diff
@@ -166,17 +170,15 @@ let dispatch_subcommand argv =
         subcmd_argv0 :: subcmd_args |> Array.of_list
       in
       let experimental = Array.mem "--experimental" argv in
+      (* basic metrics on what was the command *)
+      Metrics_.add_feature "subcommand" subcmd;
+      Metrics_.add_user_agent_tag (spf "command/%s" subcmd);
+      subcmd_argv |> Array.to_list
+      |> exclude (fun x -> not (Base.String.is_prefix ~prefix:"-" x))
```
Comment (zzeleznick) [inline src/osemgrep/cli/CLI.ml:177]: thought we might want to use the `uniq_by` operator to condense multiple usages of the `--config` flag but not picky here.

**Review by zzeleznick — APPROVED** [review]
LGTM!

File: src/osemgrep/core/Metrics_.mli (hunk 1 above)
Comment (aryx) [reply src/osemgrep/core/Metrics_.mli:6]: probably. I don't know exactly the rule either ... there's still some TODOs regarding this is_using_app

File: src/osemgrep/core/Metrics_.ml (hunk 2 above)
Comment (aryx) [reply src/osemgrep/core/Metrics_.ml:99]: yes, might be simpler. Not sure entirely what was the rational for Auto to be honest. Maybe you'll need to talk to luke or the people who wrote PRIVACY.md (maybe isaac).

File: src/osemgrep/cli/CLI.ml (hunk 3 above)
Comment (aryx) [reply src/osemgrep/cli/CLI.ml:177]: actually I don't think we should support this multiple --config xx, --config yy; I'm not sure it's very UNIXy. I think people should do --config=xx,yy instead.

**Pull request merged**